### PR TITLE
fix(avoidance): fix bug in backward length calculation

### PR DESCRIPTION
**Pull request by satoshi-ota**
### <samp>🤖 Generated by Copilot at 27610ff</samp>

Improve the avoidance module in `behavior_path_planner` by enhancing path calculation, ego index finding, and visualization methods. Use a parameter to filter out bad points in the previous path.

---

### Before this PR

Backward length is **NOT** extended properly at HIGH CURVERTURE route because it uses `calcDistance2d` to calculate distance between ego and shift start point.

```c++
  const auto longest_dist_to_shift_point = [&]() {
    double max_dist = 0.0;
    for (const auto & pnt : path_shifter_.getShiftLines()) {
      max_dist = std::max(max_dist, calcDistance2d(getEgoPose(), pnt.start));
    }
    for (const auto & sp : registered_raw_shift_lines_) {
      max_dist = std::max(max_dist, calcDistance2d(getEgoPose(), sp.start));
    }
    return max_dist;
  }();
```

269207115-07d60b50-121e-443a-a0fd-4943aa618100

### After this PR

Use `calcSignArcLength` so that it can caluclate proper distance.

```c++
  const auto longest_dist_to_shift_point = [&]() {
    double max_dist = 0.0;
    for (const auto & pnt : path_shifter_.getShiftLines()) {
      max_dist = std::max(
        max_dist, calcSignedArcLength(previous_path.points, pnt.start.position, getEgoPosition()));
    }
    for (const auto & sp : registered_raw_shift_lines_) {
      max_dist = std::max(
        max_dist, calcSignedArcLength(previous_path.points, sp.start.position, getEgoPosition()));
    }
    return max_dist;
  }();
```

269207103-cc3c98b1-a7f7-49e6-ae9f-15456cc25c96

## Tests performed

- [x] [PASS TIER IV INTERNAL SCENARIOS](https://evaluation.tier4.jp/evaluation/reports/739d42ae-fef6-541b-915c-746fe2fa5244?project_id=prd_jt)

## Effects on system behavior

Nothing.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Review by kyoichi-sugahara — APPROVED** [review]
LGTM！

File: planning/behavior_path_planner/src/scene_module/avoidance/avoidance_module.cpp [hunk 1]
```diff
@@ -1933,26 +1934,30 @@ void AvoidanceModule::generateExpandDrivableLanes(BehaviorModuleOutput & output)
 
 PathWithLaneId AvoidanceModule::extendBackwardLength(const PathWithLaneId & original_path) const
 {
+  const auto previous_path = helper_.getPreviousReferencePath();
+
   // special for avoidance: take behind distance upt ot shift-start-point if it exist.
```
Comment (kyoichi-sugahara) [inline planning/behavior_path_planner/src/scene_module/avoidance/avoidance_module.cpp:1939]: up to?

Comment (satoshi-ota) [reply planning/behavior_path_planner/src/scene_module/avoidance/avoidance_module.cpp:1939]: Oh. typo.

Comment (satoshi-ota) [reply planning/behavior_path_planner/src/scene_module/avoidance/avoidance_module.cpp:1939]: And, it's NOT special logic for avoidance  actually.

Comment (satoshi-ota) [reply planning/behavior_path_planner/src/scene_module/avoidance/avoidance_module.cpp]: Remove misreading comment in https://github.com/autowarefoundation/autoware.universe/pull/5046/commits/b6354f638f70104e5d05aab64a6c200aaadd3b7a.

**Pull request merged**